internal/controller/preservation/helper.go

```diff
@@ -32,26 +33,48 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/utils/ptr"
 
 	"github.com/stolostron/siteconfig/api/v1alpha1"
 	ci "github.com/stolostron/siteconfig/internal/controller/clusterinstance"
 )
 
+const (
+	// InternalPreservationLabelKey is used to mark resources that are required internally
+	// for ClusterInstance operations and should not be deleted during preservation or restoration.
+	InternalPreservationLabelKey = v1alpha1.PreservationLabelKey + ".internal"
+
+	// InternalPreservationLabelValue represents the label value used for identifying
+	// internally required ClusterInstance resources.
+	InternalPreservationLabelValue = "cluster-instance-required-resource"
```

Comment:
Who sets this value on resources and when? I don't see that happening in this PR anywhere, but maybe I missed it?

Reply:
The labels will be applied by a `Reinstall Handler` in a ClusterInstance controller.